microsoft/rnx-kit
# fix(rnx-build): add support for other package managers

### Description

Added support for package managers other than Yarn.

### Test plan

```sh
cd incubator/build
yarn build
yarn rnx-build --help
```

- By default, `rnx-build` should detect current package manager by looking for lock files. In `rnx-kit`, the default value for `--package-manager` should be `yarn`.
- Try adding `package-lock.json` and `pnpm-lock.yaml` to the repo root, and run `yarn rnx-build --help` again. The default value should still be `yarn`.
- Remove `yarn.lock` and run `yarn rnx-build --help`. The default value should now be `npm`.
- Remove `package-lock.json` and run `yarn rnx-build --help`. The default value should now be `pnpm`.
- Remove `pnpm-lock.yaml` and run `yarn rnx-build --help`. It should now say `required` without specifying a default value.
- Running `yarn rnx-build --platform ios` should now fail with an error message saying that `--package-manager` is required.
